src/components/v5/common/CompletedAction/partials/StreamingPayment/partials/StreamingPaymentWidget/hooks.ts

```diff
@@ -0,0 +1,53 @@
+import { BigNumber } from 'ethers';
+import { useMemo } from 'react';
+
+import { useColonyContext } from '~context/ColonyContext/ColonyContext.ts';
+import { useColonyExpenditureBalances } from '~hooks/useColonyExpenditureBalances.ts';
+import { notNull } from '~utils/arrays/index.ts';
+import { getBalanceForTokenAndDomain } from '~utils/tokens.ts';
+
+export const useHasEnoughTokensToClaim = (
+  domainId: number,
+  tokenAddress: string,
+  availableToClaim: string | null | undefined,
+) => {
+  const {
+    colony: { tokens: colonyTokens, balances },
+  } = useColonyContext();
+  const { balancesByToken: expenditureBalances, loading } =
+    useColonyExpenditureBalances();
+
+  const tokensData = useMemo(
+    () =>
+      colonyTokens?.items.filter(notNull).map((item) => {
+        const colonyTokenBalance = getBalanceForTokenAndDomain(
+          balances,
+          item.token.tokenAddress,
+          domainId,
+        );
+
+        let totalBalance = colonyTokenBalance;
+        if (domainId) {
+          const expenditureTokenBalance =
+            expenditureBalances[item.token.tokenAddress] ?? 0;
+
+          totalBalance = totalBalance.add(expenditureTokenBalance);
+        }
```

Comment:
Balances held in expenditures should not be considered here, they cannot be used when claiming a stream